src/functions.php

```diff
@@ -2,9 +2,77 @@
 
 namespace Amp\Http\Server;
 
+use Amp\ByteStream\InputStream;
 use Amp\Http\Status;
+use Amp\Loop;
+use Amp\Socket\Server as Socket;
+use Amp\Socket\SocketException;
+use Generator;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
 
 function redirectTo(string $uri, int $statusCode = Status::FOUND): Response
 {
     return new Response($statusCode, ['location' => $uri]);
 }
+
+/**
+ * Returns a response with the html content type.
+ *
+ * @param InputStream|string|null $content
+ * @param int $status
+ * @return Response
+ */
+function html($content, int $status = Status::OK): Response
+{
+    return new Response($status, [
+        'Content-Type' => 'text/html; charset=utf-8'
+    ], $content);
+}
+
+/**
+ * Creates an HTTP server.
+ *
+ * This function is inspired in Golang's http module one. It provides sensible defaults for a really quick start
+ * and it is designed to cover 90% of use cases.
+ *
+ * If you need fine-tuning, like listening in two sockets or attach more listeners to the server
+ * you should bootstrap your server manually.
+ *
+ * @param string $address
+ * @param RequestHandler $handler
+ * @param Options|null $options
+ * @param LoggerInterface|null $logger
+ * @return Generator
+ * @throws Loop\UnsupportedFeatureException
+ * @throws SocketException
+ */
+function listenAndServe(string $address, RequestHandler $handler, Options $options = null, LoggerInterface $logger = null): Generator
+{
+    $sockets = [
+        Socket::listen($address)
+    ];
+
+    $logger = $logger ?? new NullLogger();
+
+    $server = new HttpServer($sockets, $handler, $logger, $options);
+
+    yield $server->start();
+
+    // Call to server stop on SIGINT for a graceful stop
+    Loop::onSignal(SIGINT, static function ($watcherId) use ($server) {
+        Loop::cancel($watcherId);
+        yield $server->stop();
+    });
```

Comment:
I don't like the signal handler being registered here. It breaks on windows and you probably want sigint and sigterm. The signal is caught here and the server stopped, but the program doesn't exit.

Reply:
Mmm, so I need a way to pass the created `HttpServer` to the client code calling the `listenAndServe` function so they can register the signal handlers they need, but I'm unsure how to do that. Any guidance?

Reply:
@mnavarrocarter `return $server` ?

Reply:
```php
/**
 * @return Promise<HttpServer>
 */
function listenAndServe(string $address, RequestHandler $handler, Options $options = null, LoggerInterface $logger = null): Promise
{
    return call(function () use (...) {
        // your current code

        return $server;
    });
}

Reply:
Thanks, still wrapping my head around coroutines.